app/src/main/java/ch/epfl/skysync/MainActivity.kt

```diff
@@ -12,26 +12,59 @@ import androidx.navigation.compose.rememberNavController
 import ch.epfl.skysync.components.GlobalSnackbarHost
 import ch.epfl.skysync.components.SnackbarManager
 import ch.epfl.skysync.database.FirestoreDatabase
+import ch.epfl.skysync.database.UserRole
+import ch.epfl.skysync.models.user.TempUser
 import ch.epfl.skysync.navigation.MainGraph
 import ch.epfl.skysync.ui.theme.SkySyncTheme
 import com.firebase.ui.auth.FirebaseAuthUIActivityResultContract
 import com.firebase.ui.auth.data.model.FirebaseAuthUIAuthenticationResult
 import com.google.firebase.auth.FirebaseAuth
 import com.google.firebase.auth.FirebaseUser
+import kotlinx.coroutines.runBlocking
 
 class MainActivity : ComponentActivity() {
   private lateinit var signInLauncher: ActivityResultLauncher<Intent>
   private val user = mutableStateOf<FirebaseUser?>(null)
   private val db: FirestoreDatabase = FirestoreDatabase()
   private val repository: Repository = Repository(db)
 
+  private fun onError(e: Exception) {
+    println("HELLLLLOOO")
+  }
+
   private fun onSignInResult(result: FirebaseAuthUIAuthenticationResult) {
     if (result.resultCode == RESULT_OK) {
-      user.value = FirebaseAuth.getInstance().currentUser
-      SnackbarManager.showMessage("Successfully signed in")
-    } else {
-      SnackbarManager.showMessage("Authentication failed")
+      val incomingUser = FirebaseAuth.getInstance().currentUser!!
+      val email = incomingUser.email!!
+
+      runBlocking {
```

Comment:
There should not be a `runBlocking` block, instead you should define a suspend function elsewhere and wrap it in `lifecycleScope.launch`

Reply:
I will change it